### ISSUE TITLE
cjs conflict with https://deno.land/std@0.63.0/hash/mod.ts

### ISSUE THREAD

**Issue by aaronwlee** [denoland/deno#6995]
This is a strange bug.

When loading both `https://deno.land/std@0.63.0/hash/mod.ts` and some cjs files, deno checks the type of cjs.

```ts
export { createHash } from "https://deno.land/std@0.63.0/hash/mod.ts";
import {} from "./cjsTest.cjs"
```

```
error: TS2339 [ERROR]: Property 'test' does not exist on type '{}'.
  return exports.test;
                 ~~~~
    at file:///C:/Users/stone/OneDrive/%EB%B0%94%ED%83%95%20%ED%99%94%EB%A9%B4/%EC%83%88%20%ED%8F%B4%EB%8D%94/postgres/cjsTest.cjs:4:18
```

Windows 10 v2004 - X
WSL (Ubuntu bionic) - X
macOS Catalina - X

**Comment by kitsonk** [comment]
Duplicate of #6176

**Comment by aaronwlee** [comment]
@kitsonk Hi, I don't think it's been related. This cjs file loading normally working fine, but this issue happens only with this hash module.

**Comment by lucacasonato** [comment]
@aaronwlee It happens if any sibling of this cjs module is a TypeScript file, not just std/hash right? In that case it is a duplicate of #6176.

**Comment by aaronwlee** [comment]
@lucacasonato 

Sorry to confuse y'all. Ok, I think it's a more complex bug...


I thought only the combination of the hash module and cjs was a problem.
But if it's a cjs loading issue then the sample code below shouldn't work.
```ts
import { emptyDir, emptyDirSync } from "https://deno.land/std/fs/mod.ts";
// or any other modules except hash

import babelCore from "https://jspm.dev/@babel/core"; // this returns cjs.
```
Simply run this code is working perfectly fine. 
And, at this level, we can assume that babelCore is recognized as a js module rather than a cjs. 
Even if `babelCore` is a bundle of cjs files.


An unknown bug occurs in the sample code below.

This is a sample code that added a hash module.
```ts
import { emptyDir, emptyDirSync } from "https://deno.land/std/fs/mod.ts";
export { createHash } from "https://deno.land/std@0.63.0/hash/mod.ts";

import babelCore from "https://jspm.dev/@babel/core";
``` 
Run this code returns 10651 type errors...
I don't know exactly, but due to the loading of the hash module, babelCore is suddenly recognized as a party of typescript.
It's only happens with std/hash

**Comment by aaronwlee** [comment]
temp solution.

```ts
// tsconfig.json
{
  "compilerOptions": {
    "allowJs": false,
    "lib": ["ESNext", "DOM", "dom.iterable", "deno.unstable", "deno.ns"],
  }
}
```

**Comment by MichaelBauerLHIND** [comment]
I have the same Problem with the combo
import { Handlebars } from "https://deno.land/x/handlebars@v0.4.2/mod.ts";
and
import { createHash } from "https://deno.land/std@0.76.0/hash/mod.ts";

but if you use a hash module directly 
import { Sha1 } from "https://deno.land/std@0.76.0/hash/sha1.ts";
it works

maybe its a problem with the wasm module
import { Hash } from "./_wasm/hash.ts";

the tsconfig.json does't help

**Comment by kitsonk** [comment]
@MichaelBauerLHIND what version of `deno` are you using?

It appears that `https://deno.land/x/handlerbars@v0.4.2/mod.ts` has some issues independent of anything else, and probably should open an issue with the package maintain, because it is importing an untyped version of handlebars.  I am getting the following with 1.5.1:

```
error: TS2339 [ERROR]: Property 'registerHelper' does not exist on type '{}'.
        HandlebarsJS.registerHelper(helperKey, this.config.helpers[helperKey])
                     ~~~~~~~~~~~~~~
    at https://deno.land/x/handlebars@v0.4.2/mod.ts:43:22

TS2339 [ERROR]: Property 'compile' does not exist on type '{}'.
    const template = HandlebarsJS.compile(source, this.config!.compilerOptions);
                                  ~~~~~~~
    at https://deno.land/x/handlebars@v0.4.2/mod.ts:96:35

TS2339 [ERROR]: Property 'registerPartial' does not exist on type '{}'.
      HandlebarsJS.registerPartial(templateName, source);
                   ~~~~~~~~~~~~~~~
    at https://deno.land/x/handlebars@v0.4.2/mod.ts:115:20

Found 3 errors.
```

That has nothing to do with this issue.  I don't even need to import `createHash`.

The API also uses `--unstable` as well, if I omit `--unstable` I get 19 errors.  None of them related to `createHash`.

**Comment by kitsonk** [comment]
@aaronwlee I think the underlying issue is resolved in 1.5.1 (it was/is a duplicate of #6176), but there seems to be another issue related specifically the `https://jspm.dev/@babel/core` package.  There is a problem that with `allowJs` TypeScript "helpfully" tries to transpile any CommonJS modules to ESM and it is choking on a module during the emit.  I have had a little bit better luck with `https://cdn.skypack.dev/@babel/core`, but that seems to be triggering another strange TypeScript error.

**Comment by lucacasonato** [comment]
No objections to it being resolved. If you do still encounter this, please re-open.
